helix-core/src/test/java/org/apache/helix/task/TestDynamicTaskLoading.java

```diff
@@ -0,0 +1,251 @@
+    String path = TaskConstants.DYNAMICALLY_LOADED_TASK_PATH + "/Reindex";
+    removePathIfExists(path);
+    _manager.getHelixDataAccessor().getBaseDataAccessor()
+        .create(path, taskConfig.getTaskConfigZNRecord(), AccessOption.PERSISTENT);
+
+    // Submit workflow
+    TaskDriver driver = new TaskDriver(_manager);
+    String workflowName = TestHelper.getTestMethodName();
+    submitWorkflow(workflowName, driver);
+
+    try {
+      // Wait for the workflow to either complete or fail.
+      TaskState finalState =
+          driver.pollForWorkflowState(workflowName, TaskState.COMPLETED, TaskState.FAILED);
+      AssertJUnit.assertEquals(finalState, TaskState.COMPLETED);
+    } catch (HelixException e) {
+      AssertJUnit.fail(e.getMessage());
+    }
+  }
+
+  @Test
+  public void testDynamicTaskLoadingNonexistingJar() throws Exception {
+    // Add task definition information as a DynamicTaskConfig.
+    List<String> taskClasses = new ArrayList<String>();
+    taskClasses.add("com.mycompany.mocktask.MockTask");
+    DynamicTaskConfig taskConfig =
+        new DynamicTaskConfig("Reindex", "src/test/resources/Random.jar", "1.0.0", taskClasses,
+            "com.mycompany.mocktask.MockTaskFactory");
+    String path = TaskConstants.DYNAMICALLY_LOADED_TASK_PATH + "/Reindex";
+    removePathIfExists(path);
+    _manager.getHelixDataAccessor().getBaseDataAccessor()
+        .create(path, taskConfig.getTaskConfigZNRecord(), AccessOption.PERSISTENT);
+
+    // Submit workflow
+    TaskDriver driver = new TaskDriver(_manager);
+    String workflowName = TestHelper.getTestMethodName();
+    submitWorkflow(workflowName, driver);
+
+    try {
+      // Wait for the workflow to either complete or fail.
+      TaskState finalState =
+          driver.pollForWorkflowState(workflowName, TaskState.COMPLETED, TaskState.FAILED);
+      AssertJUnit.assertEquals(finalState, TaskState.FAILED);
+    } catch (HelixException e) {
+      AssertJUnit.fail(e.getMessage());
+    }
+  }
+
+  @Test
+  public void testDynamicTaskLoadingNonexistingTaskConfig() throws Exception {
+    // Remove task config ZNRecord if it exists.
+    String path = TaskConstants.DYNAMICALLY_LOADED_TASK_PATH + "/Reindex";
+    removePathIfExists(path);
+
+    // Submit workflow
+    TaskDriver driver = new TaskDriver(_manager);
+    String workflowName = TestHelper.getTestMethodName();
+    submitWorkflow(workflowName, driver);
+
+    try {
+      // Wait for the workflow to either complete or fail.
+      TaskState finalState =
+          driver.pollForWorkflowState(workflowName, TaskState.COMPLETED, TaskState.FAILED);
+      AssertJUnit.assertEquals(finalState, TaskState.FAILED);
+    } catch (HelixException e) {
+      AssertJUnit.fail(e.getMessage());
+    }
+  }
+
+  @Test
+  public void testDynamicTaskLoadingNonexistingTaskClass() throws Exception {
+    // Add task definition information as a DynamicTaskConfig.
+    List<String> taskClasses = new ArrayList<String>();
```

Comment:
Style?

Reply:
May not be related to this PR, I see a lot of Path building, like this"TaskConstants.DYNAMICALLY_LOADED_TASK_PATH + "/Reindex"  happening though out, it provides an opportunity to reduce chances of error and on the opportunity to standardize path building by defining a ZNPathBuilder utility ( This comment may be irrelevant - Sorry, am just new to the helix and trying to understand)

Reply:
Thank you for your comment. I agree this might make sense (plus your other comment regarding path building). But I guess it's more appropriate to be addresses in another PR, as you mentioned.